modules/sc-mesh-secure-deployment/src/2_0/features/cbma/macsec/macsec.py

```diff
@@ -0,0 +1,69 @@
+import subprocess
+import sys
+import os
+import random
+import threading
+
+script_dir = os.path.dirname(__file__) # path to macsec directory
+sys.path.insert(0, f'{script_dir}/../')
+from tools.custom_logger import CustomLogger
+
+logger_instance = CustomLogger("macsec")
+class Macsec:
+    def __init__(self, level, interface, macsec_encryption):
+        self.level = level # Macsec level: "lower" or "upper"
+        self.interface = interface
+        self.available_ports = set(range(1, 2**16)) # 1 to 2^16-1
+        self.used_ports = {} # client_mac: port
+        self.available_ports_lock = threading.Lock()
+        self.macsec_encryption = macsec_encryption  # Flag to set macsec encrypt on or off
+        self.logger = logger_instance.get_logger()
+
+    def set_macsec_tx(self, client_mac, my_macsec_key, my_port):
+        # Sets up macsec link and adds tx channel
+        macsec_interface = self.get_macsec_interface_name(client_mac)
+        try:
+            subprocess.run(["ip", "link", "add", "link", self.interface, macsec_interface, "type", "macsec", "port", str(my_port), "encrypt", self.macsec_encryption, "cipher", "gcm-aes-256"], check=True)
+            subprocess.run(["ip", "macsec", "add", macsec_interface, "tx", "sa", "0", "pn", "1", "on", "key", "01", my_macsec_key], check=True)
+            subprocess.run(["ip", "link", "set", macsec_interface, "up"], check=True)
+            subprocess.run(["ip", "macsec", "show"], check=True)
+            self.logger.info(f'{self.level} macsec tx channel set with {client_mac}')
+        except Exception as e:
+            self.logger.error(f'Error setting up {self.level} macsec tx channel with {client_mac}: {e}')
+            sys.exit(1)
+
+    def set_macsec_rx(self, client_mac, client_macsec_key, client_port):
+        # Adds a rx channel with client_mac, with key id = client mac without ":"
+        macsec_interface = self.get_macsec_interface_name(client_mac)
+        try:
+            subprocess.run(["ip", "macsec", "add", macsec_interface, "rx", "port", str(client_port), "address", client_mac], check=True)
+            subprocess.run(["ip", "macsec", "add", macsec_interface, "rx", "port", str(client_port), "address", client_mac, "sa", "0", "pn", "1", "on", "key", client_mac.replace(":", ""), client_macsec_key], check=True)
```

Comment:
I thought we agreed to have the peer with the smaller MAC address be the one setting Security Association 0 (alternating with 2 later when key refreshment support is added) and the one with the higher MAC setting SA 1 (alternating with 3 later) but here lines 40 and 27 both have '"sa", "0"' hardcoded so may I ask if the race condition (due to the risk of two key agreements being performed concurrently, one by ClientA & ServerB and one by ClientB & ServerA) was addressed differently?